Adding documentation for smart meter b route (#34332)
home-assistant/home-assistant.github.io: source/_integrations/smart_meter_b_route.markdown

---
title: Smart Meter B Route
description: Integrate your Smart Meter using B Route.
ha_category:
  - Energy
  - Sensor
ha_release: '2025.10'
ha_iot_class: Local Polling
ha_codeowners:
  - '@SeraphicRav'
ha_domain: smart_meter_b_route
ha_platforms:
  - sensor
ha_config_flow: true
ha_integration_type: device
---

> The Smart Meter B Route integration allows you to access the data of your smart meter using the B Route protocol.

Reviewer: ```suggestion
The **Smart Meter B Route** {% term integration %} allows you to access the data of your smart meter using the B Route protocol.
```

> The Smart Meter B Route integration allows you to access the data of your smart meter using the B Route protocol.

Reviewer: Can you add a link to an explainer of the B route protocol? How do I know this integration is for me? Do I need to be with a special provider? have a special smart meter?

## Overview

This integration enables you to access and monitor your household energy consumption data from a smart meter using the B Route protocol in Home Assistant. You'll need a compatible USB dongle and your B Route credentials to set up this integration. It has been tested for the Japanese market.

## Prerequisites

1. **Compatible USB Dongle**: Tested with the Wi-SUN USB dongle from [Ratoc Systems](https://www.ratocsystems.com/products/wisun/usb-wisun/rs-wsuha/).
2. **B Route Credentials**: Obtain your B Route credentials from your utility company.
3. **Smart Meter**: A smart meter that supports the B Route protocol. If you are renting, you will need to contact your house management company, as the installation is free but may require permission.

{% include integrations/config_flow.md %}

> ## Signal Reach

Reviewer: ```suggestion
## Signal reach
```
We use [sentene-style capitalization](https://developers.home-assistant.io/docs/documenting/general-style-guide), also in headings and UI strings. Please fix throughout the page.

The Wi-SUN USB dongle has a typical signal reach of approximately 50 meters indoors and up to 200 meters outdoors, depending on environmental factors such as walls, interference, and other obstacles.

> ## Obtaining B Route Credentials

Reviewer: This section is part of the prereqs. I would move it there.

> 1. **Compatible USB dongle**: Tested with the Wi-SUN USB dongle from [Ratoc Systems](https://www.ratocsystems.com/products/wisun/usb-wisun/rs-wsuha/).
> 2. **B Route credentials**: Obtain your B Route credentials from your utility company.
> 3. **Smart meter**: A smart meter that supports the B Route protocol. If you are renting but do not have a smart meter equipped, you will need to contact your house management company, as the installation is free but may require permission.

Reviewer: ```suggestion
1. Compatible USB dongle: 
    - Tested with the Wi-SUN USB dongle from [Ratoc Systems](https://www.ratocsystems.com/products/wisun/usb-wisun/rs-wsuha/).
2. [B Route credentials from your utility company](#optaining-b-route-credentials).
3. Smart meter: 
    - A smart meter that supports the B Route protocol. 
    - If you are renting but do not have a smart meter equipped, you will need to contact your house management company, as the installation is free but may require permission.
```
Because bold is used to markup UI strings, we avoid using it to highlight text, if possible.

Ensure that your power supply is not a bulk power supply for an apartment building.

Refer to the links below to apply for your B Route credentials:

- [Hokkaido Electric Power](https://www.hepco.co.jp/network/electric_life/service/electronic_meter/b_route_service_low.html)
- [Tohoku Electric Power](https://nw.tohoku-epco.co.jp/consignment/request/other/)
- [TEPCO](http://www.tepco.co.jp/pg/consignment/liberalization/smartmeter-broute.html)
- [Chubu Electric Power](https://www.chuden.co.jp/home/smartmeter/intro/use/index.html)
- [Hokuriku Electric Power](http://www.rikuden.co.jp/nw_kojin/b_routeservice.html)
- [Kansai Electric Power](http://www.kepco.co.jp/corporate/smartmeter/routeb/routeb_how.html)
- [Chugoku Electric Power](https://www.energia.co.jp/nw/safety/smartmeter/route-b.html)
- [Shikoku Electric Power](https://www.yonden.co.jp/nw/b_root/index.html)
- [Kyushu Electric Power](https://www.kyuden.co.jp/td_service_meter_b-root_index.html)
- [Okinawa Electric Power](https://www.okiden.co.jp/business-support/service/smartmeter/b-route/index.html)

> ## Get your monthly consumption data

Reviewer: ```suggestion
## Retrieving your monthly consumption data
```

By default, the data you get is the total consumption since an unknown date in the past. If you want to get your monthly consumption data, you can use the [utility_meter](/integrations/utility_meter/) {% term integration %}.

> ## Get your monthly cost data

Reviewer: ```suggestion
## Retrieving your monthly cost data
```

> To get your monthly cost data, you can use the [template sensor](/integrations/template/) {% term integration %} along with the [utility_meter](/integrations/utility_meter/) {% term integration %}. You will need to know your electricity rate (cost per kWh) to calculate the cost. For instance, if you are using Tokyo Gas and have a price depending on your usage, you can use the following template with a unit of measurement set to your currency per kWh (e.g., JPY/kWh) and the appropriate sensor name:

Reviewer: ```suggestion
To get your monthly cost data, you can use the [template sensor](/integrations/template/) {% term integration %} along with the [utility_meter](/integrations/utility_meter/) {% term integration %}. You will need to know your electricity rate (cost per kWh) to calculate the cost. For instance, if you are using Tokyo Gas and have a price depending on your usage, you can use the following template with a unit of measurement set to your currency per kWh (for example, JPY/kWh) and the appropriate sensor name:
```
